File: Solutionsin Haskell/p009.hs
{- 
 - Solution to Project Euler problem 9
 - Copyright (c) Sankha Saha (CODR). All rights reserved.
 - 
 - https://github.com/codr07/Project-Euler-Solusions
 -}


-- Computers are fast, so we can implement a brute-force search to directly solve the problem.
perim = 1000
main = putStrLn (show ans)
ans = head [a * b * (perim - a - b) | a <- [1..perim], b <- [a+1..perim], isIntegerRightTriangle a b]
isIntegerRightTriangle a b = a < b && b < c
	&& a * a + b * b == c * c
	where c = perim - a - b

File: codes/mathematica/p009.m
(* 
 * Solution to Project Euler problem 9
 * Copyright (c) Sankha Saha. All rights reserved.
 * 
 *  
 *   
 *)


(* Computers are fast, so we can implement a brute-force search to directly solve the problem. *)
For[a = 1, a <= 1000, a++,
  For[b = a + 1, b <= 1000, b++,
    Block[{c = 1000 - a - b},
      If[a < b < c && a^2 + b^2 == c^2,
        Print[a, " ", b, " ", c, " ", a * b * c]]]]]
